flask/app/tasks.py

```diff
@@ -1,50 +1,34 @@
 import os
 from datetime import datetime, timedelta
+from itertools import chain
+
 from . import models
 from .email import send_email
 from .extensions import db
-from sqlalchemy import text
 
 
 def send_email_notification(app):
-    app.logger.info("hello")
     with app.app_context():
+        yesterday = (datetime.now() - timedelta(1)).strftime("%Y-%m-%d")
 
-        app.logger.info("Sending email notification...")
-
-        yesterday = (datetime.now() - timedelta(5)).strftime("%Y-%m-%d")
-
-        analyses = db.session.query(models.Analysis).filter(
-            models.Analysis.requested >= yesterday,
+        datasets = (
+            db.session.query(models.Dataset)
+            .join(models.Dataset.analyses)
+            .filter(models.Analysis.requested >= yesterday)
+            .all()
         )
 
-        app.logger.info("Available analyses...")
-        app.logger.debug(analyses)
+        analyses = list(chain(*[d.analyses for d in datasets]))
```

Comment:
I think `chain.from_iterable` will flatten the list of lists without this notation? Like
```python
for analysis in chain.from_iterable(d.analyses for d in datasets):
```
This is only needed if the query is performed on the dataset rather than analysis table though.